network/endpoint_windows.go

```diff
@@ -14,16 +14,58 @@ import (
 	"github.com/Microsoft/hcsshim"
 )
 
+// ConstructEpName constructs endpoint name from netNsPath.
+func ConstructEpName(containerID string, netNsPath string, ifName string) string {
+	epName := ""
+	if netNsPath != "" {
+		splits := strings.Split(netNsPath, ":")
+		if len(splits) == 2 {
+			epName = splits[1]
+		} else {
+			epName = containerID
+		}
+		if len(epName) > 8 {
+			epName = epName[:8] + "-" + ifName
```

Comment:
Why are we choosing last 8 characters of ContainerID? Will the last 8 characters will be unique for all containers?

Reply:
We construct endpoint ID using first 8 characters of containerID and interface name. Please check plugin.go line 162-168.